packages/diagram-render-svg/src/svgRenderer.ts

```diff
@@ -169,33 +215,73 @@ class SVGDiagramVisitor extends SimplifiedDiagramVisitor<undefined, SVGNode[]> {
         return [result];
     }
 
-    override visitText(element: SimplifiedText): SVGNode[] {
-        const result: SVGNode = {
+    override visitText(element: SimplifiedText, context: SVGRendererContext): SVGNode[] {
+        return [
+            this.textAsPath ? this.renderTextAsPath(context, element) : this.renderTextAsText(element),
+            ...this.visitChildren(element, context)
+        ];
+    }
+
+    /**
+     * Renders a text element as path
+     *
+     * @param context the context to use
+     * @param element the text element to render
+     * @returns the svg node representing the text
+     */
+    private renderTextAsPath(context: SVGRendererContext, element: SimplifiedText): SVGNode {
+        const font = context.getFont(element.fontFamily);
+        const glyphRun = font.layout(element.text);
+        let offset = 0;
+        const paths: string[] = [];
+        for (const glyph of glyphRun.glyphs) {
+            if (!/^\s*$/.test(String.fromCodePoint(...glyph.codePoints)) && element.fill != undefined) {
+                paths.push(glyph.path.translate(offset, 0).toSVG());
+            }
+            offset += glyph.advanceWidth;
```

Comment:
Is it intentional that `offset` is global for the entire glyph run while it seems to be accessed locally for each glyph?
If yes, I guess this means that each glyph can perform arithmetic relative to the parent?
Otherwise this implementation makes no sense.

Reply:
this just means that each glyph is located after the previous one, starting at offset 0